pkg/pillar/cmd/nodeagent/nodeagent.go

```diff
@@ -613,6 +618,32 @@ func handleLastRebootReason(ctx *nodeagentContext) {
 	ctx.lastLock.Unlock()
 }
 
+// handleInstallationLog checks if first-boot file from installation exists
+// remove it and send log from installation to the controller
+func handleInstallationLog(_ *nodeagentContext) {
+	if fileExists(firstbootInstallFile) {
+		err := os.Remove(firstbootInstallFile)
```

Comment:
If we power off we'd potentiall miss some log entries. Better to do the remove at the end (means if power off we will repeat some logs which is better).

Reply:
I put small timeout before removing of the file to be on the safe side here.
Also I slightly reworked logic in device-steps to move the file from the installer (first-boot) into another file (send-require) to not assume twice that it was the first boot after installer.